sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -2092,11 +2098,247 @@ def set_application(self, channel, appl_code):
+        '''
+        This function applies RX output amp settings
+        '''
+        for lane in range(self.NUM_CHANNELS):
+            if ((1 << lane) & host_lanes_mask) == 0:
+                continue
+            lane = lane+1
+            si_key_lane = "{}{}".format(consts.OUTPUT_AMPLITUDE_TARGET_RX, lane)
+            val = si_settings[consts.OUTPUT_AMPLITUDE_TARGET_RX][si_key_lane]
+            if val <= get_rx_output_eq_post_max_val():
+                self.nibble_read_modify_write(val, consts.OUTPUT_AMPLITUDE_TARGET_RX, lane)
+
+    def stage_fixed_input_target_tx(self, host_lanes_mask, si_settings):
+        '''
+        This function applies fixed TX input si settings
+        '''
+        for lane in range(self.NUM_CHANNELS):
+            if ((1 << lane) & host_lanes_mask) == 0:
+                continue
+            lane = lane+1
+            si_key_lane = "{}{}".format(consts.FIXED_INPUT_EQ_TARGET_TX, lane)
+            val = si_settings[consts.FIXED_INPUT_EQ_TARGET_TX][si_key_lane]
+            if val <= get_rx_output_eq_post_max_val():
+                self.nibble_read_modify_write(val, consts.FIXED_INPUT_EQ_TARGET_TX, lane)
+
+    def stage_adaptive_input_recall_tx(self, host_lanes_mask, si_settings):
+        '''
+        This function applies adaptive TX input recall si settings
+        '''
+        val = 0
+        for lane in range(self.NUM_CHANNELS):
+            if ((1 << lane) & host_lanes_mask) == 0:
+                continue
+            si_key_lane = "{}{}".format(consts.ADAPTIVE_INPUT_EQ_ENABLE_TX, lane+1)
+            si_val = si_settings[consts.ADAPTIVE_INPUT_EQ_ENABLE_TX][si_key_lane]
+            lane %= (self.NUM_CHANNELS//2)
+            mask = ~(val << (lane*2))
+            l_data = si_val << (lane*2)
+            val = (val & mask) | l_data
+            self.xcvr_eeprom.write(si_key_lane, val)
+
+    def byte_read_modify_write(self, si_keys, host_lanes_mask, si_settings):
+        val = 0
+        # Read 1 byte data
+        for lane in range(self.NUM_CHANNELS):
+            si_key_lane = "{}{}".format(si_keys, lane+1)
+            data  = self.xcvr_eeprom.read(si_key_lane)
+            val |= (data << lane)
+        # Write only applicable field
+        for lane in range(self.NUM_CHANNELS):
+            if ((1 << lane) & host_lanes_mask) == 0:
+                continue
+            si_key_lane = "{}{}".format(si_keys, lane+1)
+            si_val = si_settings[si_keys][si_key_lane]
+            val &= ~(1 << lane)
+            val |= (si_val << lane)
+            self.xcvr_eeprom.write(si_key_lane, val)
+
+    def stage_adaptive_input_eq_enable_tx(self, host_lanes_mask, si_settings):
+        '''
+        This function applies adaptive TX input enable si settings
+        '''
+        self.byte_read_modify_write(consts.ADAPTIVE_INPUT_EQ_ENABLE_TX, host_lanes_mask, si_settings)
+
+    def stage_cdr_tx(self, host_lanes_mask, si_settings):
```

Comment:
follow same for [tx_disable_channel](https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_xcvr/api/public/cmis.py#L621)?

Reply:
Hi @prgeor, Thanks for the comments. Can I take care of these changes in next commit? I have already finished most of the UT with existing changes. I will create a Jira issue to track this.

Reply:
Taken care. Since we are reading the SCS0, I am using to to apply applicable nibble value to byte data read using EEPROM read. Logic reference from tx_disable_channel.